apps/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -31,31 +34,66 @@ export class PackageChangeAnalyzer {
     this._git = new Git(this._rushConfiguration);
   }
 
-  public async getPackageDeps(
+  /**
+   * Try to get a list of the specified project's dependencies and their hashes.
+   *
+   * @remarks
+   * If the data can't be generated (i.e. - if Git is not present), this throws an Error.
+   * If the project name is invalid, this returns undefined.
+   */
+  public async getProjectDependenciesAsync(
+    projectName: string,
+    terminal: Terminal
+  ): Promise<Map<string, string> | undefined> {
+    if (this._data === null) {
+      this._data = await this._getDataAsync(terminal);
+    }
+
+    if (this._data === undefined) {
+      throw new Error('Unable to get current repo state.');
```

Comment:
Seems a bit arbitrary that:
- `undefined` is returned if the caller misspells the project name , which is an actual  mistake
- an exception is thrown if the `.git` folder doesn't exist, which commonly occurs if your project isn't tracked by Git yet